### PR TITLE
[READY FOR REVIEW] Adds Github's CI via Actions

### PR DESCRIPTION
**Pull request by Shadow-Quill**
# This is generally untested.
With so many files touched, it's fairly difficult for one person to test everything. I don't *believe* anything has been fucked, though a testmerge may be prudent.

## About The Pull Request

This PR *will* fail its own checks. Reasons being:
- Unformatted maps. This will be done in a later PR.
- Incorrect pixel offset in map(s). Same as above.
- Integration Tests will fail because discord.dm is failing. @vulppine was poked about this, and I was told he'll fix it Soon™.

also, any changes to core files were done upstream as well. (and I'm pretty sure I didn't do any custom edits myself)
- Note: I've slightly modified some sections so Integration Test logs are colored. They've been appropriately commented.
## Why It's Good For The Game

CI ensures that 2+2=4 in game (amongst other stuff, if a test has been written for it), and otherwise makes sure that a given PR can actually compile.

# Please merge this with the squash-type merge.

### CODE REVIEW

**Comment by Shadow-Quill** [comment]
Yes, I'll squash my commits once I'm done.

**Comment by Shadow-Quill** [comment]
As an additional note, [Github gives us Actions for free](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions) because we're open source.

File: code/world.dm [hunk 1]
```diff
@@ -17,4 +17,288 @@
 	view = "15x15"
 	hub = "Exadv1.spacestation13"
 	cache_lifespan = 0	//stops player uploaded stuff from being kept in the rsc past the current session
-	fps = 20
```
Comment (Shadow-Quill) [inline code/world.dm]: note to self, check this against code/game/world.dm
changes here duped there, need to fix.

File: code/game/world.dm [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+	if(!dbcon || !dbcon.IsConnected())
+		return setup_database_connection()
+	else
+		return 1
+
+/world/proc/incrementMaxZ()
+	maxz++
+	SSmobs.MaxZChanged()
+
+/world/proc/change_fps(new_value = 30)
```
Comment (Michiyamenotehifunana) [inline code/game/world.dm:393]: I just came here to make sure this is here

Comment (Shadow-Quill) [reply code/game/world.dm:393]: ?

**Review by Jamini — APPROVED** [review]
Test merge soon(tm)

**Pull request merged**